encoders/firebase-encoders/src/main/java/com/google/firebase/encoders/json/JsonDataEncoderBuilder.java

```diff
@@ -32,7 +34,7 @@
 import java.util.Map;
 import java.util.TimeZone;
 
-public final class JsonDataEncoderBuilder {
+public final class JsonDataEncoderBuilder implements EncoderConfig<JsonDataEncoderBuilder> {
```

Comment:
I'm not sure this class **is** a configuration, but rather, it **has** a configuration. We can encapsulate the object & value encoders below into a configuration object and pass that to the configurator instead of the whole builder being a config.

Reply:
wdyt?

Reply:
> I'm not sure this class is a configuration, but rather, it has a configuration

I guess one could argue either way but in my view when I use a builder to create an object I am configuring/describing the properties I want the object to have, so in this sense I feel like builder is a configuration.

I guess we could have the builder contain a config object inside, but it seems that then the builder will not do anything other than delegate to the contained config object, i.e. it will have `registerEncoder` and `configureWith` methods which will just call into the config, which will effectively mean that the builder has all the methods that `EncoderConfig` requires but does not implement `EncoderConfig` itself.

Reply:
I think the builder would need to receive the `EncoderConfig` object the client would generate itself. The Builder wouldn't need to expose the methods itself. That being said I'm not strongly attached to that design. I'll approve the PR and you can go ahead with the changes.